code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,752 @@
+	icon_state = "SLRW-rocket"
+	ex_range = 4
+
+/obj/item/projectile/a66mm/emp/on_hit(turf/target)
+	if(locate(/obj/mecha) in target.contents)
+		for(var/obj/mecha/M in target.contents)
+			if(!is_fellow_antag(M) && M.density)
+				M.take_damage(damage, BRUTE)
+				break
+	else if(locate(/mob/living) in target.contents)
+		for(var/mob/living/L in target.contents)
+			if(!is_fellow_antag(L) && L.density)
+				L.apply_damage(damage, BRUTE)
+				L.check_projectile_dismemberment(src, "chest")
+				break
+	else
+		target = src
+
+	if(friendly_fire_dud(target, ex_range))
+		return FALSE
+
+	empulse(target, -1, ex_range, TRUE, cause = "[type] fired by [key_name(firer)]")
+	new/obj/effect/temp_visual/emp/pulse(target)
+	exsfx_hit(ex_range+4, ex_range+10, 'sound/effects/empulse.ogg')
+	ffx_hit(target)
+
+/obj/item/projectile/a66mm/incen
+	name ="66mm Incendiary rocket"
+	desc = "A rocket designed to ignite targets in a wide area."
+	icon_state = "SLRW-rocket"
+	ex_range = 3
+	var/list/fire_stop = null // List of types that would stop the spread of a fire ball.
+
+/obj/item/projectile/a66mm/incen/Initialize()
+	..()
+	fire_stop = list()
+	fire_stop += /turf/unsimulated/wall
+	fire_stop += /turf/simulated/wall
+	fire_stop += /turf/space
+	fire_stop += /obj/structure/shuttle
+	fire_stop += /obj/structure/window/plasmabasic
+	fire_stop += /obj/structure/window/full/plasmabasic
+	fire_stop += /obj/structure/window/plasmareinforced
+	fire_stop += /obj/structure/window/full/plasmareinforced
+
+/obj/item/projectile/a66mm/incen/on_hit(turf/target)
+	if(locate(/mob/living) in target.contents)
+		for(var/mob/living/L in target.contents)
+			if(!is_fellow_antag(L) && L.density)
+				L.apply_damage(damage, BRUTE)
+				L.check_projectile_dismemberment(src, "chest")
+				break
+	else
+		target = get_turf(src)
+
+	for(var/i in fire_stop)
+		if(istype(target, i))
+			return FALSE
+
+		if(locate(i) in target.contents)
+			return FALSE
+
+	if(friendly_fire_dud(target, ex_range))
+		return FALSE
+
+	for(var/turf/T in range(ex_range, target))
+		var/set_fire = TRUE
+		var/turf/walker = get_turf(target)
+		var/dist = hypotenuse(T.x, T.y, walker.x, walker.y)
+		if(dist > ex_range)
+			continue
+		while(walker != T && set_fire)
+			walker = get_step_towards(walker, T)
+			for(var/i in fire_stop)
+				if(istype(walker, i))
+					set_fire = FALSE
+					break
+				if(locate(i) in walker.contents)
+					set_fire = FALSE
+					break
+			if(!set_fire)
+				break
+			for(var/obj/O in walker.contents)
+				if(istype(O, /obj/structure/window))
+					var/obj/structure/window/W = O
+					W.take_damage(W.obj_integrity)
+				if(istype(O, /obj/structure/grille))
+					var/obj/structure/window/W = O
+					W.take_damage(W.obj_integrity)
+				if(istype(O, /obj/machinery/door))
+					if(O.density)
+						set_fire = FALSE
+						break
+
+		if(set_fire)
+			new /obj/effect/hotspot(T)
+			ffx_hit(T)
+			for(var/mob/living/L in T)
+				L.EyeBlurry(10)
```

Comment:
Can't this blurr the shit out of bystanders?

Reply:
It should only blur mobs in a hotspot tile.

Reply:
Ah yep you're right. Only on that tile. I misinterpreted it